WebGoatCore/Views/Product/AddOrEdit.cshtml

```diff
@@ -16,19 +16,26 @@
         <legend>Product information</legend>
         @if (!Model.AddsNew)
         {
-            @Html.HiddenFor(m => m.Product.ProductId)
-            @Html.HiddenFor(m => m.Product.SupplierId)
+            #pragma warning disable CS8602 // False positive, cannot happen in template usage
+            Html.HiddenFor(m => m.Product.ProductId);
+            Html.HiddenFor(m => m.Product.SupplierId);
+            #pragma warning restore CS8602
             <p>
+                @{ #pragma warning disable CS8602 } <!-- False positive, cannot happen in template usage -->
```

Comment:
Why one location needs `@{ #pragma` both other uses simply `#pragma`?

Reply:
@{} can be omitted if we are in context of C# block. In other places we are inside html tags.

Reply:
Both of them are in the same context I believe

Reply:
I was wrong, sorry